src/shared/getNpmUserInfo.ts

```diff
@@ -0,0 +1,39 @@
+import { $ } from "execa";
+import npmUser, { type UserInfo } from "npm-user";
+
+export type { UserInfo };
+export interface ErrorResult {
+	reason: string;
+	succeeded: false;
+}
+export interface SuccessResult {
+	succeeded: true;
+	value: UserInfo;
+}
+type Result = ErrorResult | SuccessResult;
+
+export async function getNpmUserInfo(): Promise<Result> {
+	let username;
+
+	try {
+		const { stdout } = await $`npm whoami`;
+
+		username = stdout;
+	} catch {
+		return {
+			reason: "Could not populate npm user. Failed to run npm whoami.",
+			succeeded: false,
+		};
+	}
+
+	try {
+		const user = await npmUser(username);
+		return { succeeded: true, value: user };
+	} catch {
```

Comment:
[Non-Actionable] 😠 why is Codecov saying these aren't covered by tests? They clearly are...!

> **Codecov** / codecov/patch
> src/shared/getNpmUserInfo.ts#L32
>
> Added line #L32 was not covered by tests

There's something wrong with (most likely: how I've configured) Codecov... your unit tests are excellent and thorough and beautiful as they are.